das/daser.go

```diff
@@ -71,9 +56,16 @@ func NewDASer(
 		subscriber:     newSubscriber(),
 		subscriberDone: make(chan struct{}),
 	}
-	d.sampler = newSamplingCoordinator(concurrencyLimit, samplingRange, getter, d.sample)
 
-	return d
+	for _, applyOpt := range options {
+		err := applyOpt(d)
```

Comment:
To avoid duplication of errors in options and Validate on Params, let's remove errors from Options and call here `Validate` after applying all the options. This way we use Validate twice on the `moddaser` and inside the Daser constructor here, but we avoid duplication which good. Suggested by @walldiss and I think this is good idea and hope that's the last fix before we can say our configuration pattern is complete